src/ActualLab.Rpc/Infrastructure/RpcInterceptor.cs

```diff
@@ -33,9 +33,17 @@ public RpcInterceptor(
         var rpcMethodDef = (RpcMethodDef)methodDef;
         var localCallAsyncInvoker = methodDef.SelectAsyncInvokerUntyped(initialInvocation.Proxy, LocalTarget);
         return invocation => {
-            Task resultTask;
             var context = RpcOutboundCallSetup.ProduceContext();
-            var call = context.PrepareCall(rpcMethodDef, invocation.Arguments);
+            RpcOutboundCall? call;
+            try {
+                call = context.PrepareCall(rpcMethodDef, invocation.Arguments);
+            }
+            catch (RpcRerouteException) {
+                var rerouteResultTask = InvokeWithRerouting(rpcMethodDef, context, null, localCallAsyncInvoker, invocation);
```

Comment:
I'll check this piece, but RpcRerouteException shouldn't be thrown at that phase.

Reply:
Check how wrapper method for routing works, it automatically handles "rerouting on routing" cases.